test/unit/product/product-updater.js

```diff
@@ -34,108 +39,218 @@ describe('Product Updater class', () => {
-          tracker: {
-            trackMethod: (fn) => {
-              return function(...params) {
-                fn(...params);
-              };
-            },
-          },
-        }));
-      },
-      createModal() {
-        return new Modal(config, props);
+        return Promise.resolve(new Cart(config, {}));
       },
-      closeModal: closeModalSpy,
     };
+    wrapper = document.createElement('div');
     sinon.stub(props.client.shop, 'fetchInfo').resolves(shopFixture);
     sinon.stub(props.client.product, 'fetch').resolves(productFixture);
     configCopy = Object.assign({}, config);
     configCopy.node = document.createElement('div');
     configCopy.node.setAttribute('id', 'fixture');
     document.body.appendChild(configCopy.node);
-    testProductCopy = Object.assign({}, productFixture);
     product = new Product(configCopy, props);
+    product.view.wrapper = wrapper;
+    initStub = sinon.stub(product, 'init');
+    superUpdateConfigStub = sinon.stub(Updater.prototype, 'updateConfig');
   });
 
   afterEach(() => {
     document.body.removeChild(configCopy.node);
+    initStub.restore();
+    superUpdateConfigStub.restore();
   });
 
   describe('updateConfig()', () => {
-    const newConfig = {
-      options: {
-        modalProduct: {
-          layout: 'vertical',
-        },
-      },
-    };
-
-    beforeEach(async () => {
-      await product.init(testProductCopy);
-    });
-
-    it('calls updateConfig on cart', () => {
-      const cartUpdateConfigSpy = sinon.stub(product.cart, 'updateConfig');
-      product.updateConfig(newConfig);
-      assert.calledWith(cartUpdateConfigSpy, newConfig);
-    });
-
-    it('calls updateConfig on modal if modal exists', async () => {
-      const modalProduct = new Product({
-        node: configCopy.node,
-        options: Object.assign({}, configCopy.options, {
-          product: Object.assign({}, configCopy.options.product, {
-            buttonDestination: 'modal',
-          }),
-        }),
-      }, props);
-      await modalProduct.init(testProductCopy);
-      await modalProduct.openModal();
-      const cartUpdateConfigSpy = sinon.spy();
-      modalProduct.cart = {
-        updateConfig: cartUpdateConfigSpy,
-      };
-      const modalUpdateConfigSpy = sinon.spy(modalProduct.modal, 'updateConfig');
-      modalProduct.updateConfig(newConfig);
-      assert.calledWith(modalUpdateConfigSpy, sinon.match.object);
-      assert.equal(modalProduct.modal.config.product.layout, 'vertical');
-      assert.calledWith(cartUpdateConfigSpy, newConfig);
-      modalUpdateConfigSpy.restore();
-      modalProduct.modal.destroy();
-    });
-
-    describe('when updating ID, storefront ID, variant ID, or storefront variant ID', () => {
-      let initSpy;
-
-      beforeEach(() => {
-        initSpy = sinon.spy(product, 'init');
-      });
-
-      it('calls init if ID updated', () => {
+    describe('if storefrontId or storefrontVariantId is supplied from config', () => {
```

Comment:
I think technically normal ids can be specified in the config, the `normalizeConfig` just transforms them into their respective storefront version. Maybe we could say `if id or variant id is supplied in config`, to be generic enough to encompass normal and storefront ids